### ISSUE TITLE
How to customize the distribution tables

### ISSUE THREAD

**Issue by martinholmer** [TPRU-demo/pitaxcalc-demo#84]
In merged pull request #80, @sebastiansajie asked the following questions:
> I ran the code for the distribution tables and it works well. However as there were too many columns, I outputted the tables to a csv file to be able to take a closer look.
> `dt1.to_csv('Distribution Tables_Base.csv')`
> `dt2.to_csv('Distribution Tables_Reform.csv')`
>
> I have a couple of questions:
>
> 1. We also need the average burden per taxpayer. I would be able to divide each of the columns in the csv file with the weight and arrive at it. It would be good to have this as an option. No big deal though if this is not done as we need to work with the csv files in any case for reporting.
> 2. Would be possible to select specific columns (variables) into the tables?  This would allow us to see specific columns while running the program.

Part of the problem is that the table cell amounts have not been scaled and rounded.  It seems to me that it would make sense to have money values expressed in billions and rounded to three decimal places and head-count values expressed in millions and rounded to three decimal places.  I'll add that as an option to the distribution_tables method.   When I do that I'll also add an option to show weighted averages instead of aggregate totals.

I'm not sure what you mean by "for reporting" when you say:
> we need to work with the csv files in any case for reporting

If you want to show model results to senior people, a `csv` file is not easy to look at.
Instead of using the `to_csv()` DataFrame method, try the `to_string()` DataFrame method.
Here is what the whole (unscaled) distribution table looks like when using the `to_string()` method:

<img width="1674" alt="screen shot 2018-12-14 at 5 15 33 pm" src="https://user-images.githubusercontent.com/12170745/50034111-5424b000-ffc9-11e8-94f1-962dda8d4f81.png">

Of course, it would be narrower and much easier to read if the amounts were scaled.

In terms, of selecting a subset of columns to show, both the `to_csv()` and `to_string()` methods have a `columns` argument that accepts a list of column names that you want included in the table.  Here is some code that produces a subtable:
```
import taxcalc as tc

# create current-law Calculator object calc1
rec = tc.Records()
pol = tc.Policy()
calc1 = tc.Calculator(policy=pol, records=rec)
assert calc1.current_year == 2017
calc1.calc_all()

# create policy-reform Calculator object calc2
reform = {2017: {'_rate2': [0.06]}}  # reform lowers rate from 10% to 6%
pol.implement_reform(reform)
calc2 = tc.Calculator(policy=pol, records=rec)
calc2.calc_all()
assert calc2.current_year == 2017

# create decile distribution tables for current law and policy reform
dt1, dt2 = calc1.distribution_tables(calc2, 'weighted_deciles')

# print raw version of distribution tables to screen
print('Current-Law Distribution Table for 2017 (raw df)')
print(dt1)
print('Policy-Reform Distribution Table for 2017 (raw df)')
print(dt2)

# print text version of each complete distribution table to a file
with open('dist-table-all-clp.txt', 'w') as dfile:
    dt1.to_string(dfile)
with open('dist-table-all-ref.txt', 'w') as dfile:    
    dt2.to_string(dfile)

# print text version of each partial distribution table to a file
to_include = ['weight', 'GTI', 'TTI', 'pitax']
with open('dist-table-part-clp.txt', 'w') as dfile:
    dt1.to_string(dfile, columns=to_include)
with open('dist-table-part-ref.txt', 'w') as dfile:    
    dt2.to_string(dfile, columns=to_include)
```
And here is what the `dist-table-part-clp.txt` file looks like:

<img width="580" alt="screen shot 2018-12-14 at 5 30 31 pm" src="https://user-images.githubusercontent.com/12170745/50034290-196f4780-ffca-11e8-98c9-8d20175b4ff3.png">

So, I don't think it makes sense to add the subtable option to the `distribution_tables` method because you can achieve that with just a line or two of code.   But the other two options (scaling/rounding and weighted averages) are more involved, and therefore, should be added as options to the `distribution_tables` method.  I'll work on adding those two options in the coming days.

**Comment by sebastiansajie** [comment]
@martinholmer This is helpful. On the 'reporting' that I was referring to, the csv is useful so that non-python experts could use Excel charts to put in their reports. We could also i,port the .txt file into Excel, so your suggestion works too.

**Comment by martinholmer** [comment]
@sebastiansajie,  pull request #85 is an attempt to resolve issue #84.
If pull request makes sense, then merge it into the master branch.
If you have concerns or questions, I'll be back in the office on January 5th.

**Comment by martinholmer** [comment]
Merged pull request #85 resolves issue #84.
